Users/abehenderson/CSX/Recursion/factorial.js

```diff
@@ -0,0 +1,34 @@
+// Attempt #2 (CSX Solution)
+
+function factorial(num, product = 1) {
+  if (num === 1) return product;
+  return factorial(num - 1, product * num);
+}
+
+console.log(factorial(4)); // -> 24
+console.log(factorial(6)); // -> 720
+
+// I don't understand this solution very well.  It seems unneccesarily complicated.  Is there something I don't know?
+
+//--------------------------------------------------------------------
+// Attempt #1
+
+function factorial(num) {
+  return (num <= 1) ? num : num * factorial(num - 1);
+}
+
+// or..
+
+const factorial = (num => (num <= 1) ? num : num * factorial(num - 1));
```

Comment:
It's definitely good to try different things but in my mind this is a fail: the conditional part of your ternary here is a bit of technical mumbo-jumbo and should be avoided.  There's no difference functionally between what you had in attempt 1 `(num <= 1)` and here: `num => (num <= 1)`, it's just adding a bit of complication; also, `factorial` isn't actually a function, so this doesn't meet the requirements of the problem - technically here `factorial` is the evaluated result of an anonymous function.
Why is that distinction important? Because when you start interviewing, you have to solve the problem you're given, rather than a slightly different one, and you must meet the requirements in the problem statement.

Reply:
"`factorial` isn't actually a function, so this doesn't meet the requirements of the problem - technically here `factorial` is the evaluated result of an anonymous function."

Ok, yeah that makes a lot of sense.  I guess in this case I was playing with it just to see how succinct I could make it, but in doing so I fundamentally changed `factorial` from a function to the evaluated result of an anonymous function...like you said.